eo-maven-plugin/src/main/java/org/eolang/maven/DiscoverMojo.java

```diff
@@ -94,26 +95,33 @@ public void exec() throws IOException {
     private Collection<String> discover(final Path file)
         throws FileNotFoundException {
         final XML xml = new XMLDocument(file);
+        final List<String> base = new ListOf<>(
+            "//o[",
+            "not(starts-with(@base,'.'))",
+            "and @base != 'Q'",
+            "and @base != '^'",
+            "and @base != '$'",
+            "and @base != '&'",
+            "and not(@ref)"
+        );
+        final List<String> regular = new ListOf<>(base);
+        if (this.versioned) {
+            regular.add("and not(@ver)");
+        }
+        regular.add("]/@base");
         final Collection<String> names = new TreeSet<>(
-            new ListOf<>(
-                new Filtered<>(
-                    obj -> !obj.isEmpty(),
-                    xml.xpath(
-                        String.join(
-                            " ",
-                            "//o[",
-                            "not(starts-with(@base,'.'))",
-                            " and @base != 'Q'",
-                            " and @base != '^'",
-                            " and @base != '$'",
-                            " and @base != '&'",
-                            " and not(@ref)",
-                            "]/@base"
-                        )
-                    )
-                )
-            )
+            this.names(xml, regular)
         );
+        if (this.versioned) {
+            final List<String> versioned = new ListOf<>(base);
```

Comment:
@maxonfjvipon `this.versioned` and just `versioned` are rather confusing names (you have the same name for both variables). Maybe it's better to use different name for one of them?